process/block/alteredaccounts/alteredAccountsProvider.go

```diff
@@ -151,17 +158,13 @@ func (aap *alteredAccountsProvider) addTokensDataForMarkedAccount(
 	storageKey := []byte(core.ElrondProtectedKeyPrefix + core.ESDTKeyIdentifier)
 	storageKey = append(storageKey, tokenKey...)
 
-	esdtTokenBytes, err := userAccount.RetrieveValueFromDataTrieTracker(storageKey)
-	if err != nil {
-		log.Warn("cannot get ESDT data for address",
-			"address", encodedAddress,
-			"tokenKey", tokenKey,
-			"error", err)
+	userAccountVmCommon, ok := userAccount.(vmcommon.UserAccountHandler)
+	if !ok {
+		log.Warn("state.UserAccountHandler cannot be cast to vmcommon.UserAccountHandler", "address", aap.addressConverter.Encode(userAccount.GetOwnerAddress()))
 		return nil
```

Comment:
why not returning error here?

Reply:
you're right. it's a programming error. returned error + added unit test

Reply:
:hat-off2: